tests/codegen/llvm/kfunc_dereference.ll

```diff
@@ -6,7 +6,7 @@ target triple = "bpf-pc-linux"
 ; Function Attrs: nounwind
 declare i64 @llvm.bpf.pseudo(i64 %0, i64 %1) #0
 
-define i64 @"kfunc:vmlinux:tcp_sendmsg"(i8* %0) section "s_kfunc:vmlinux:tcp_sendmsg_1" {
+define i64 @"kfunc:tcp_sendmsg"(i8* %0) section "s_kfunc:tcp_sendmsg_1" {
```

Comment:
Any idea why this is happening? I've encountered the same using the old `update_codegen_tests.sh` on my machine. It seems that the function -> module mapping is not computed correctly and IIRC, it should be taken from `/sys/kernel/debug/tracing/available_filter_functions`.

Maybe it's the same reason as with the other changed codegen test?

Reply:
Took a look. It's cuz codegen tests looked at the host's real tracefs for available_filter_functions. It "worked" in CI cuz codegen tests run as root in the container and we bind mount in debugfs. But locally it silently fails. The correct fix is to mock out the access (perhaps more correct is to use fakes and dependency inject, but a different discussion).

I've put up the fix in #2712